lib/lita/adapters/slack/api.rb

```diff
@@ -188,20 +183,27 @@ def connection
             unless config.proxy.nil?
               options = { proxy: config.proxy }
             end
-            Faraday.new(options)
+            retry_options = {
+              retry_statuses: [429]
+            }
+            Faraday.new(options) do |f|
+              f.request :url_encoded
+              f.request :retry, retry_options
+              f.adapter Faraday.default_adapter
+            end
           end
         end
 
         def parse_response(response, method)
-          unless response.status == 429 || response.success?
+          unless response.success?
             raise "Slack API call to #{method} failed with status code #{response.status}: '#{response.body}'. Headers: #{response.headers}"
           end
 
           MultiJson.load(response.body)
         end
 
         def fetch_cursor(page)
-          page.dig("response_metadata", "next_cursor")
+          page.dig("response_metadata", "next_cursor") || ""
```

Comment:
Any reason why you're returning an empty string here? I'm not sure it really changes any of the above behavior.

Reply:
A lot of the tests in spy don't mock this value, and it might not be present in the responses from slack (all the paginated ones should have it, but based on the state of the documentation, i would rather it be safe)

And, unless you have a newer version of active support, .empty? Isn't on nil values, so it broke a lot of things in spy